python/ray/util/check_serialize.py

```diff
@@ -0,0 +1,205 @@
+            for name, obj in closure.globals.items():
+                serializable, _ = inspect_serializability(
+                    obj,
+                    name=name,
+                    parent=parent,
+                    depth=depth - 1,
+                    _failure_set=failure_set)
+                found = found or not serializable
+                if found:
+                    break
+
+    if closure.nonlocals:
+        _printer.print(
+            f"Detected {len(closure.nonlocals)} nonlocal variables. "
+            "Checking serializability...")
+        with _printer.indent():
+            for name, obj in closure.nonlocals.items():
+                serializable, _ = inspect_serializability(
+                    obj,
+                    name=name,
+                    parent=parent,
+                    depth=depth - 1,
+                    _failure_set=failure_set)
+                found = found or not serializable
+                if found:
+                    break
+    if not found:
+        _printer.print(
+            f"WARNING: Did not find non-serializable object in {base_obj}. "
+            "This may be an oversight.")
+    return found
+
+
+def _inspect_generic_serialization(base_obj, depth, parent, failure_set):
+    assert not inspect.isfunction(base_obj)
+    functions = inspect.getmembers(base_obj, predicate=inspect.isfunction)
+    found = False
+    with _printer.indent():
+        for name, obj in functions:
+            serializable, _ = inspect_serializability(
+                obj,
+                name=name,
+                parent=parent,
+                depth=depth - 1,
+                _failure_set=failure_set)
+            found = found or not serializable
+            if found:
+                break
+
+    with _printer.indent():
+        members = inspect.getmembers(base_obj)
+        for name, obj in members:
+            if name.startswith("__") and name.endswith(
+                    "__") or inspect.isbuiltin(obj):
+                continue
+            serializable, _ = inspect_serializability(
+                obj,
+                name=name,
+                parent=parent,
+                depth=depth - 1,
+                _failure_set=failure_set)
+            found = found or not serializable
+            if found:
+                break
+    if not found:
+        _printer.print(
+            f"WARNING: Did not find non-serializable object in {base_obj}. "
+            "This may be an oversight.")
+    return found
+
+
+def inspect_serializability(base_obj,
+                            name=None,
+                            parent=None,
+                            depth=3,
+                            _failure_set=None):
+    """Identifies what objects are preventing serialization.
+
+    Args:
+        base_obj: Object to be serialized.
+        name (str): Optional name of string.
+        depth (int): Depth of the scope stack to walk through. Defaults to 3.
```

Comment:
`parent ` and `_failure_set ` are not documented here.

Reply:
Moved them to private - not planning to document private variables.